src/StepProgress.tsx

```diff
@@ -84,36 +128,27 @@ export function StepProgress({
           <FontAwesomeIcon icon={faChevronLeft} />
         </button>
       )}
-      {elements.map((element, ix) => (
-        <button
-          className={twMerge(
-            'flex flex-1 items-center justify-center p-1 hover:bg-primary-20 hover:text-primary',
-            ix === 0 && 'rounded-l',
-            ix === length - 1 && 'rounded-r',
-            value > ix && !items && 'bg-primary-60 text-white',
-            value === ix && 'bg-gray-400 font-bold text-white',
-            element.status === 'correct' &&
-              'bg-green-600 bg-opacity-60 text-white',
-            element.status === 'incorrect' &&
-              'bg-red-600 bg-opacity-60 text-white',
-            element.status === 'partial' &&
-              'bg-uzh-red-100 bg-opacity-60 text-white',
-            value === ix && 'bg-opacity-100',
-            displayOffset && ix < value - displayOffset && 'hidden',
-            displayOffset && ix > value + displayOffset && 'hidden'
-          )}
-          onClick={() => onItemClick(ix, items && items[ix])}
-        >
-          {element.status === 'correct' && (
-            <FontAwesomeIcon icon={faCheckDouble} />
-          )}
-          {element.status === 'incorrect' && <FontAwesomeIcon icon={faX} />}
-          {element.status === 'partial' && <FontAwesomeIcon icon={faCheck} />}
-          {(!element.status || element.status === 'unanswered') && (
-            <div>{ix + 1}</div>
-          )}
-        </button>
-      ))}
+      {elements.map((element, ix) => {
+        const formattedElement = formatter({ element, ix })
+        return (
+          <button
+            className={twMerge(
+              'flex flex-1 items-center justify-center border-r border-white p-1 last:border-r-0 hover:bg-primary-20 hover:text-primary',
+              ix === 0 && 'rounded-l',
+              ix === length - 1 && 'rounded-r',
+              value > ix && !items && 'bg-primary-60 text-white',
+              value === ix && 'bg-gray-400 font-bold text-white',
+              displayOffset && ix < value - displayOffset && 'hidden',
+              displayOffset && ix > value + displayOffset && 'hidden',
+              formattedElement.className,
+              value === ix && 'bg-opacity-100'
```

Comment:
Should we maybe move the `formattedElement.className` behind the `value === ix && "bg-opacity-100"` line to be able to override this as well if desired?